library/common/main_interface.cc

```diff
@@ -21,6 +21,19 @@ static std::shared_ptr<Envoy::Engine> engine() {
   return engine_.lock();
 }
 
+bool runOnEngineDispatcher(envoy_engine_t,
+                           std::function<void(std::shared_ptr<Envoy::Engine>)> func) {
```

Comment:
The function should pass the Engine by ref (`Envoy::Engine&`), not by shared pointer, to avoid lifetime extension